src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEM.hpp

```diff
@@ -0,0 +1,426 @@
+
+  } waveEquationViewKeys;
+
+
+protected:
+
+  virtual void postProcessInput() override final;
+
+  virtual void initializePostInitialConditionsPreSubGroups() override final;
+
+private:
+
+  /**
+   * @brief Locate sources and receivers position in the mesh elements, evaluate the basis functions at each point and save them to the
+   * corresponding elements nodes.
+   * @param mesh mesh of the computational domain
+   * @param regionNames the names of the region you loop on
+   */
+  virtual void precomputeSourceAndReceiverTerm( MeshLevel & mesh, arrayView1d< string const > const & regionNames ) override;
+
+  /**
+   * @brief Apply free surface condition to the face define in the geometry box from the xml
+   * @param time the time to apply the BC
+   * @param domain the partition domain
+   */
+  virtual void applyFreeSurfaceBC( real64 const time, DomainPartition & domain ) override;
+
+  /**
+   * @brief Initialize Perfectly Matched Layer (PML) information
+   */
+  virtual void initializePML() override;
+
+  /**
+   * @brief Apply Perfectly Matched Layer (PML) to the regions defined in the geometry box from the xml
+   * @param time the time to apply the BC
+   * @param domain the partition domain
+   */
+  virtual void applyPML( real64 const time, DomainPartition & domain ) override;
+
+
+  /// save the sismo trace in file
+  void saveSeismo( localIndex iseismo, real64 val, string const & filename ) override;
+
+  /// Indices of the nodes (in the right order) for each source point
+  array2d< localIndex > m_sourceNodeIds;
+
+
+  /// Constant part of the source for the nodes listed in m_sourceNodeIds in x-direction
+  array2d< real64 > m_sourceConstantsx;
+
+  /// Constant part of the source for the nodes listed in m_sourceNodeIds in x-direction
+  array2d< real64 > m_sourceConstantsy;
+
+  /// Constant part of the source for the nodes listed in m_sourceNodeIds in x-direction
+  array2d< real64 > m_sourceConstantsz;
+
+  /// Flag that indicates whether the source is local or not to the MPI rank
+  array1d< localIndex > m_sourceIsLocal;
+
+  /// Indices of the element nodes (in the right order) for each receiver point
+  array2d< localIndex > m_receiverNodeIds;
+
+  /// Basis function evaluated at the receiver for the nodes listed in m_receiverNodeIds
+  array2d< real64 > m_receiverConstants;
+
+  /// Flag that indicates whether the receiver is local or not to the MPI rank
+  array1d< localIndex > m_receiverIsLocal;
+
+  /// Displacement_np1 at the receiver location for each time step for each receiver (x-component)
+  array2d< real64 > m_displacementxNp1AtReceivers;
+
+  /// Displacement_np1 at the receiver location for each time step for each receiver (y-component)
+  array2d< real64 > m_displacementyNp1AtReceivers;
+
+  /// Displacement_np1 at the receiver location for each time step for each receiver (z-component)
+  array2d< real64 > m_displacementzNp1AtReceivers;
+
+};
+
+
+namespace extrinsicMeshData
```

Comment:
The treatment of the `extrinsicMeshData` in the wave propagation solvers (elastic and acoustic) has diverged a little bit from the rest of the code. We don't include the `EXTRINSIC_MESH_DATA_TRAIT` in the hpp file of the solver anymore because this tends to introduce circular dependencies in the code.

For instance, below you define `Displacementx_n` in `ElasticWaveEquationSEM.hpp`. Then in `ElasticWaveEquationSEMKernels.hpp` you use `Displacementx_n` in the kernel here:
https://github.com/GEOSX/GEOSX/blob/5ef6490a44bc9070c7db170c078521808687caaa/src/coreComponents/physicsSolvers/wavePropagation/ElasticWaveEquationSEMKernel.hpp#L562
This means that the kernel code has to know about the solver code, which is something that we trying to avoid.

So in the rest of the code, we create a separate file, named `NameOfTheSolverExtrinsicData.hpp` that contains all the macros that you have defined below. Then we include `NameOfTheSolverExtrinsicData.hpp` both in `NameOfTheSolver.hpp` and in `NameOfTheSolverKernels.hpp`. 

We also define a namespace for each type of physics, so we have for instance `extrinsicMeshData::flow::pressure`. You can find an example [in this file](https://github.com/GEOSX/GEOSX/blob/develop/src/coreComponents/physicsSolvers/fluidFlow/FlowSolverBaseExtrinsicData.hpp).

Reply:
Okay I note that for our refactor, thanks François